modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/JBossProductDiscovery.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * RHQ Management Platform
+ * Copyright (C) 2005-2015 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+package org.rhq.modules.plugins.jbossas7;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class JBossProductDiscovery {
+    private static final Log LOG = LogFactory.getLog(JBossProductDiscovery.class);
+    private static final Map<String, JBossProduct> KNOWN_PRODUCTS = new HashMap<String, JBossProduct>();
+
+    static {
+        Iterator<JBossProduct> products = Arrays.asList(
+            JBossProduct.AS,
+            JBossProduct.EAP,
+            JBossProduct.WILDFLY8,
+            JBossProduct.ISPN,
+            JBossProduct.JDG,
+            JBossProduct.JPP,
+            JBossProduct.SOA,
+            JBossProduct.JDV,
+            JBossProduct.unknown()
```

Comment:
Are you sure this how discovery callbacks works? My understanding was callbacks were called after the plugin container got the PluginConfig details from AS7 plugin.